app/src/main/java/ca/marshallasch/veil/utilities/Util.java

```diff
@@ -168,17 +186,48 @@ public static String generateHash(byte[] data) {
      *
      * @param message the body of the comment
      * @param author the author who wrote the comment
+     * @param anonymous if the post is anonymous or not
+     * @return a comment object with the postID field unset
+     */
+    public static DhtProto.Comment createComment(@NonNull String message, @NonNull DhtProto.User author, boolean anonymous) {
+
+        return createComment(message, author, null, anonymous);
+    }
+
+    /**
+     * Create a new comment object, that is not assigned to a specific post.
+     * Note that this does not set the postID field, and the hash is made with the field unset.
+     * This will set anonymous to false.
+     *
+     * @param message the body of the comment
+     * @param author the author who wrote the comment
      * @param postHash the postID that it belongs to
      * @return a comment object with the postID field set
      */
     public static DhtProto.Comment createComment(@NonNull String message, @NonNull DhtProto.User author, @Nullable String postHash) {
 
+        return createComment(message, author, postHash, false);
+    }
+
+    /**
+     * Create a new comment object, that is not assigned to a specific post.
+     * Note that this does not set the postID field, and the hash is made with the field unset.
+     *
+     * @param message the body of the comment
+     * @param author the author who wrote the comment
+     * @param postHash the postID that it belongs to
+     * @param anonymous if the post is anonymous or not
+     * @return a comment object with the postID field set
+     */
+    public static DhtProto.Comment createComment(@NonNull String message, @NonNull DhtProto.User author, @Nullable String postHash, boolean anonymous) {
```

Comment:
Maybe write some tests to see if the expected result will be set for more edge case scenarios especially involving the 'anonymous' param.